088-rebalance-progress-status

```diff
@@ -0,0 +1,328 @@
+`
+```
+[1] An example of the ExecutorState JSON of an inter-broker partition rebalance. 
+
+### Progress Update Cadence
+
+For ease of implementation and minimizing the load on the CruiseControl REST API server, we would only query the CruiseControlState endpoint and update the “progress” section upon `KafkaRebalance` resource reconciliation. 
+The progress section will never be more out of date longer than the reconciliation period and even if the rebalance runs into an error or “NotReady” state, the “progress” section would still be updated on that KafkaRebalance resource reconciliation along with any error.
+
+### Future Improvements
+
+#### Adding “progress” section for other KafkaRebalance states
+
+In addition to the “progress” the “Rebalancing” and “Stopped” KafkaRebalance states, we could provide the “progress” section for other states as well such as the “ProposalReady” and “Ready” states. 
+Firstly, this would help emphasize that a rebalance had not started or had completed by having a percentageComplete: 0% on "ProposalReady" and a percentageComplete: 100% on "Ready". 
+This emphasis could help clear up ambiguity surrounding what the KafkaRebalance “Ready” state or “optimizationResult” field means. 
+Secondly, but more importantly, it would provide an estimate for the minimum time a partition rebalance proposal would take to execute before even executing it. 
+This feature would be of great value to users. 
+However, providing an accurate estimation for this is non-trivial, namely the “estimatedTimeToCompletion” field for “ProposalReady" state, is non-trivial. 
+
+Leveraging the Cruise Control configurations and user-provided network capacity settings, we could provide a rough estimate for “estimatedTimeToCompletetion” field for inter-broker balances. 
+However, one challenge is coming up with a method of reliably determining a reasonable estimate for the disk read/write throughput. 
+It is not so much of an issue for inter-broker rebalance estimates (assuming network is the bottleneck for inter-broker balances) but is certainly an issue for intra-broker rebalance estimates. 
+
+Estimation for inter-broker partition rebalance time:
+```
+# The maximum number of partition movements given CC partition movement cap
+max_partition_movements= min(<# of brokers> *  
+    num.concurrent.partition.movements.per.broker) 
+max_partition_movements=min(max_partition_movements, max.num.cluster.partition.movements)
```

Comment:
Why are these two variables called the same thing?

Reply:
I was wondering the same thing, like are we overwriting the previous value? I think a different name would be better.

Reply:
Fixed with the reformating